### PR TITLE
Remove /tasks/{taskId}/ack endpoint

### PR DESCRIPTION
**Pull request by balloonio**
Discussed with @kishorebanala in #1501 , the `/tasks/{taskId}/ack` endpoint is no longer needed and to be removed.

In this PR, I only removed the endpoint on the API layer, as I didn't want to prematurely remove anything from the core. Please let me know if more removal or deprecation should be included in this PR.

### CODE REVIEW

**Comment by balloonio** [comment]
@kishorebanala  Bumping up the thread

**Comment by kishorebanala** [comment]
Hey @balloonio, Thanks for your contribution. As I was mentioning in #1501, we cannot remove this API in minor version upgrade, as this would break several clients, especially the previous versions which still calls the ack API.

For now, we shall mark this API deprecated, and do the same in client implementation as well.

**Comment by christianscheuer** [comment]
A quick question as a new user to the current state of things with the /ack endpoint.
I'm seeing the following odd behaviors right now:
- /ack endpoint always returns false
- /poll endpoint returns the same task more than once when called concurrently from different worker threads (from C# client for example)
- Python client calls ack and only proceeds if ack returns true, so it never perform any actions
- C# client runs the ack endpoint but always proceeds regardless of the ack result.

So in other words - right now the C# client ends up processing the task more than once and the python client ends up not processing anything at all.

Since the /ack endpoint is deprecated, shouldn't the /poll endpoint always only return the task exactly once?
And shouldn't /ack always return true for backwards compatibility with clients like the python client that will only proceed if it returns true?

**Comment by kishorebanala** [comment]
Closing due to inactivity, please reopen if you are able to resume working on this.
